sirepo/sim_api/jupyterhublogin.py

```diff
@@ -33,15 +34,15 @@
 
 
 class API(sirepo.api.Base):
-    @sirepo.api_perm.allow_visitor
+    @sirepo.api_perm.manual_permission_check
     def api_checkAuthJupyterhub(self):
         def _res_for_uri(uri):
             return self.reply_ok(PKDict(uri=uri))
 
         u = None
         try:
             sirepo.auth.require_user()
-            sirepo.auth.require_sim_type('jupyterhublogin')
+            sirepo.auth.check_sim_type_role('jupyterhublogin')
```

Comment:
I think this is a better method to use:

`self.parse_params(type='jupyterhublogin')`

This is part of our problem. I think if this were written with a, say, sim_type parameter like other calls e.g. downloadFile, then you wouldn't need to hardwire the type, which probably should come from a constant in the module, just for cleanliness.

Reply:
890681e6a4c0784a71559f1a88c2d2665ec78e19